File: designer/types/reaction.ts
import { XOneOrArray } from "./core";
import { XAction } from "./action";

export interface IReaction {
	command: string;
	inputs?: IReactionInput[];
	actions: XOneOrArray<XAction>;
}

export interface IReactionInput {
	input: string;
	value: string;
}

File: api/BoningerWorks.TextAdventure.Json/Inputs/Reaction.cs
using BoningerWorks.TextAdventure.Json.Utilities;
using System.Text.Json.Serialization;

namespace BoningerWorks.TextAdventure.Json.Inputs
{
	public class Reaction
	{
		[JsonPropertyName("command")] public string? CommandId { get; set; }
		[JsonPropertyName("inputs")] public OneOrManyList<Input?>? Inputs { get; set; }
		[JsonPropertyName("actions")] public OneOrManyList<Action?>? Actions { get; set; }
	}
}
